src/internal/packager/images/progress.go

```diff
@@ -0,0 +1,133 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package images provides image-related utilities for Zarf
+package images
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log/slog"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/zarf-dev/zarf/src/pkg/utils"
+	"oras.land/oras-go/v2"
+)
+
+// Report defines a function to log progress
+type Report func(bytesRead, totalBytes int64)
+
+// DefaultReport returns a default report function
+func DefaultReport(l *slog.Logger, msg string, imageName string) Report {
+	return func(bytesRead, totalBytes int64) {
+		percentComplete := float64(bytesRead) / float64(totalBytes) * 100
+		remaining := float64(totalBytes) - float64(bytesRead)
+		l.Info(msg, "name", imageName, "complete", fmt.Sprintf("%.1f%%", percentComplete), "remaining", utils.ByteFormat(remaining, 2))
+	}
+}
+
+const defaultProgressInterval = 5 * time.Second
+
+// StartReporting starts the reporting goroutine
```

Comment:
What happens if StartReporting is called twice? is there any potential state that needs to be tracked?

Reply:
I think that is fair - my brain picks up on tickers for potential memory leaks but I remembered that this was resolved in [go 1.23](https://go.dev/doc/go1.23#timer-changes)